packages/components/src/internal/actions.ts

```diff
@@ -2013,143 +1743,111 @@ async function prepareInsertRowDataFromBulkForm(
     };
 }
 
-export function pasteEvent(
-    modelId: string,
+export async function pasteEvent(
+    editorModel: EditorModel,
+    dataKeys: List<any>,
+    data: Map<any, Map<string, any>>,
+    queryInfo: QueryInfo,
     event: any,
-    onBefore?: any,
-    onComplete?: any,
     columnMetadata?: Map<string, EditableColumnMetadata>,
-    readonlyRows?: List<any>,
+    readonlyRows?: List<string>,
     lockRowCount?: boolean
-): void {
-    const model = getEditorModel(modelId);
-
+): Promise<EditorModelAndGridData> {
     // If a cell has focus do not accept incoming paste events -- allow for normal paste to input
-    if (model && model.hasSelection() && !model.hasFocus()) {
+    if (editorModel && editorModel.hasSelection() && !editorModel.hasFocus()) {
         cancelEvent(event);
-        pasteCell(
-            modelId,
-            model.selectedColIdx,
-            model.selectedRowIdx,
-            getPasteValue(event),
-            onBefore,
-            onComplete,
+        const value = getPasteValue(event);
+        return await pasteCell(
+            editorModel,
+            dataKeys,
+            data,
+            queryInfo,
+            value,
             columnMetadata,
             readonlyRows,
             lockRowCount
         );
     }
+
+    return { data: undefined, dataKeys: undefined, editorModel: undefined };
 }
 
 async function pasteCell(
-    modelId: string,
-    colIdx: number,
-    rowIdx: number,
-    value: any,
-    onBefore?: any,
-    onComplete?: any,
+    editorModel: EditorModel,
+    dataKeys: List<any>,
+    data: Map<any, Map<string, any>>,
+    queryInfo: QueryInfo,
+    value: string,
```

Comment:
We were using them, and they were only ever used to show/hide the mask after you paste, while we're loading data. This was because the original pasteCell was not an async function or a promise, even though it used promises. This made it impossible for the EditableGrid to know when paste was complete, so we had to pass in the callbacks so this function could do it. Now that it's async EditableGrid can `await` the function and show/hide the mask before/after calling.